canonical-web-and-design/maas-ui
# feat(ui): allow adding storage to existing datastores

## Done

- Added "Add to existing datastore" to VMFS6 bulk actions

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a machine with at least 2 disks, and is in Ready or Allocated state. (If you set the MAAS to karura you can find one at http://0.0.0.0:8400/MAAS/r/machine/8qcbef/storage).
- Change the storage layout to VMFS6, which will automatically partition and use one of the disks
- Partition the other disk into 3 unformatted partitions
- Select one of the partitions and create a datastore from it
- Select both of the remaining partitions and click "Add to existing datastore"
- Check that the total size is correct
- Check that you can choose from the 2 newly created datastores, and the mount point updates correctly
- Check that you can successfully add storage to a datastore

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2172
